File: 3950-exactly-one-consecutive-set-bits-pair/3950-exactly-one-consecutive-set-bits-pair.py
class Solution:
    def consecutiveSetBits(self, n: int) -> bool:
        count = 0
        prev = 0

        while n:
            curr = n & 1

            if curr == 1 and prev == 1:
                count += 1
                if count > 1:
                    return False

            prev = curr
            n >>= 1

        return count == 1

File: 3950-exactly-one-consecutive-set-bits-pair/3950-exactly-one-consecutive-set-bits-pair.java
class Solution {
    public boolean consecutiveSetBits(int n) {
        int count = 0;
        while(n>0){
            if((n&3)==3){
                count++;
                if(count>1)
                {
                    return false;
                }
            }
            n>>=1;
        }
        return count==1;
    }
}